src/Traits/BooleanTypeTrait.php

```diff
@@ -39,33 +38,36 @@ trait BooleanTypeTrait
      * Create a conjunction between this expression and the given expression.
      *
      * @param BooleanType $right
+     * @param bool $insertParentheses
      * @return AndOperator
      */
-    public function and(BooleanType $right): AndOperator
+    public function and(BooleanType $right, bool $insertParentheses = true): AndOperator
```

Comment:
The `$insertParentheses` parameter should also be added to the relevant interface (`BooleanType`).